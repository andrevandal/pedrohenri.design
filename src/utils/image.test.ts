import { describe, it, expect } from 'vitest'
import {
  getImageUrl,
  parseSizes,
  generateSrcSet,
  generateSizesString,
  generateImageProps
} from '@utils/image'

describe('getImageUrl', () => {
  it('should return the correct URL', () => {
    const url = getImageUrl('images/footer.png', 'f_auto,q_90,w_540')
    expect(url).toMatchSnapshot()
  })
})

describe('parseSizes', () => {
  it('should parse the sizes with default fallback', () => {
    const sizes = 'xs:540px sm:640px lg:1024px xl:1280px'
    const parsedSizes = parseSizes(sizes)
    expect(parsedSizes).toEqual({
      default: '1280px',
      xs: '540px',
      sm: '640px',
      lg: '1024px',
      xl: '1280px'
    })
  })

  it('should parse the with default fallback (smaller than xs)', () => {
    const sizes = 'xs:540px sm:300px lg:1024px xl:1280px'
    const parsedSizes = parseSizes(sizes)
    expect(parsedSizes).toEqual({
      default: '1280px',
      xs: '540px',
      sm: '300px',
      lg: '1024px',
      xl: '1280px'
    })
  })

  it('should parse the sizes with all screen', () => {
    const sizes = '540px sm:640px lg:1024px xl:1280px'
    const parsedSizes = parseSizes(sizes)
    expect(parsedSizes).toEqual({
      default: '540px',
      sm: '640px',
      lg: '1024px',
      xl: '1280px'
    })
  })
})

describe('generateSrcSet', () => {
  it('should generate the srcSet', () => {
    const src = 'images/footer.png'
    const sizes = 'xs:540px sm:640px lg:1024px xl:1280px'
    const srcSet = generateSrcSet(src, parseSizes(sizes))
    expect(srcSet).toMatchSnapshot()
  })
})

describe('generateSizesString', () => {
  it('should generate the sizes string', () => {
    const sizes = 'xs:540px sm:640px lg:1024px xl:1280px'
    const sizesString = generateSizesString(parseSizes(sizes))
    expect(sizesString).toBe(
      '(min-width: 1280px) 1280px, (min-width: 1024px) 1024px, (min-width: 640px) 640px, (min-width: 480px) 540px, 1280px'
    )
  })
})

describe('generateImageProps', () => {
  it('should generate the image props', () => {
    const src = 'images/footer.png'
    const sizes = 'xs:540px sm:640px lg:1024px xl:1280px'
    const imageProps = generateImageProps(src, sizes)
    expect(imageProps).toMatchSnapshot()
  })
})
