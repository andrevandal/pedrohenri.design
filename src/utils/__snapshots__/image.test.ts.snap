// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateImageProps > should generate the image props 1`] = `
{
  "sizes": "(min-width: 1280px) 1280px, (min-width: 1024px) 1024px, (min-width: 640px) 640px, (min-width: 480px) 540px, 1280px",
  "src": "https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png",
  "srcSet": "https://ipx.vandal.services/f_auto,q_90,w_540/https://files.vandal.services/pedrohenri.design/images/footer.png 540w, https://ipx.vandal.services/f_auto,q_90,w_810/https://files.vandal.services/pedrohenri.design/images/footer.png 810w, https://ipx.vandal.services/f_auto,q_90,w_1080/https://files.vandal.services/pedrohenri.design/images/footer.png 1080w, https://ipx.vandal.services/f_auto,q_90,w_1350/https://files.vandal.services/pedrohenri.design/images/footer.png 1350w, https://ipx.vandal.services/f_auto,q_90,w_1620/https://files.vandal.services/pedrohenri.design/images/footer.png 1620w, https://ipx.vandal.services/f_auto,q_90,w_640/https://files.vandal.services/pedrohenri.design/images/footer.png 640w, https://ipx.vandal.services/f_auto,q_90,w_960/https://files.vandal.services/pedrohenri.design/images/footer.png 960w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1600/https://files.vandal.services/pedrohenri.design/images/footer.png 1600w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_1024/https://files.vandal.services/pedrohenri.design/images/footer.png 1024w, https://ipx.vandal.services/f_auto,q_90,w_1536/https://files.vandal.services/pedrohenri.design/images/footer.png 1536w, https://ipx.vandal.services/f_auto,q_90,w_2048/https://files.vandal.services/pedrohenri.design/images/footer.png 2048w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3072/https://files.vandal.services/pedrohenri.design/images/footer.png 3072w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3200/https://files.vandal.services/pedrohenri.design/images/footer.png 3200w, https://ipx.vandal.services/f_auto,q_90,w_3840/https://files.vandal.services/pedrohenri.design/images/footer.png 3840w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3200/https://files.vandal.services/pedrohenri.design/images/footer.png 3200w, https://ipx.vandal.services/f_auto,q_90,w_3840/https://files.vandal.services/pedrohenri.design/images/footer.png 3840w",
}
`;

exports[`generateSrcSet > should generate the srcSet 1`] = `"https://ipx.vandal.services/f_auto,q_90,w_540/https://files.vandal.services/pedrohenri.design/images/footer.png 540w, https://ipx.vandal.services/f_auto,q_90,w_810/https://files.vandal.services/pedrohenri.design/images/footer.png 810w, https://ipx.vandal.services/f_auto,q_90,w_1080/https://files.vandal.services/pedrohenri.design/images/footer.png 1080w, https://ipx.vandal.services/f_auto,q_90,w_1350/https://files.vandal.services/pedrohenri.design/images/footer.png 1350w, https://ipx.vandal.services/f_auto,q_90,w_1620/https://files.vandal.services/pedrohenri.design/images/footer.png 1620w, https://ipx.vandal.services/f_auto,q_90,w_640/https://files.vandal.services/pedrohenri.design/images/footer.png 640w, https://ipx.vandal.services/f_auto,q_90,w_960/https://files.vandal.services/pedrohenri.design/images/footer.png 960w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1600/https://files.vandal.services/pedrohenri.design/images/footer.png 1600w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_1024/https://files.vandal.services/pedrohenri.design/images/footer.png 1024w, https://ipx.vandal.services/f_auto,q_90,w_1536/https://files.vandal.services/pedrohenri.design/images/footer.png 1536w, https://ipx.vandal.services/f_auto,q_90,w_2048/https://files.vandal.services/pedrohenri.design/images/footer.png 2048w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3072/https://files.vandal.services/pedrohenri.design/images/footer.png 3072w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3200/https://files.vandal.services/pedrohenri.design/images/footer.png 3200w, https://ipx.vandal.services/f_auto,q_90,w_3840/https://files.vandal.services/pedrohenri.design/images/footer.png 3840w, https://ipx.vandal.services/f_auto,q_90,w_1280/https://files.vandal.services/pedrohenri.design/images/footer.png 1280w, https://ipx.vandal.services/f_auto,q_90,w_1920/https://files.vandal.services/pedrohenri.design/images/footer.png 1920w, https://ipx.vandal.services/f_auto,q_90,w_2560/https://files.vandal.services/pedrohenri.design/images/footer.png 2560w, https://ipx.vandal.services/f_auto,q_90,w_3200/https://files.vandal.services/pedrohenri.design/images/footer.png 3200w, https://ipx.vandal.services/f_auto,q_90,w_3840/https://files.vandal.services/pedrohenri.design/images/footer.png 3840w"`;

exports[`getImageUrl > should return the correct URL 1`] = `"https://ipx.vandal.services/f_auto,q_90,w_540/https://files.vandal.services/pedrohenri.design/images/footer.png"`;
